python/ray/util/options.py

```diff
@@ -0,0 +1,202 @@
+"""Manage, parse and validate options for Ray tasks, actors and actor methods."""
+from typing import Dict, Any, Callable, Tuple, Union, Optional
+from dataclasses import dataclass
+from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy
+
+_SENTINEL = object()
+
+
+@dataclass
+class Option:
+    # Type constraint of an option.
+    type_constraint: Optional[Union[type, Tuple[type]]] = None
+    # Value constraint of an option.
+    value_constraint: Optional[Callable[[Any], bool]] = None
+    # Error message.
+    error_message: Optional[str] = None
+    # Default value.
+    default_value: Any = _SENTINEL
+    # Default value factory.
+    default_value_factory: Optional[Callable] = None
+    # This option is only valid with ".options()"
+    options_only: bool = False
```

Comment:
Currently only `max_calls` is `remote_only`, I am not sure if it is over design.

Reply:
I feel this is cleaner conceptually: an option can be remote only, options only and both. We don't need to make exceptions somewhere else for max_calls.

Reply:
I further checked the code, it seems that all options allowed in `.options()` is equally valid for `@ray.remote`, it is more likely that we did not support it because people did not add update `@ray.remote` to sync with new options. I will solidify this choice by asking people related to API changes.

Also only `max_calls` and `max_concurrency` are `@ray.remote` only, so I decide remove `options_only` and `remote_only` at the same time.